src/plugins/tonegen/interact.c

```diff
@@ -0,0 +1,187 @@
+/*************************************************************************
+This file is part of tone-generator
+
+Copyright (C) 2010 Nokia Corporation.
+
+This library is free software; you can redistribute
+it and/or modify it under the terms of the GNU Lesser General Public
+License as published by the Free Software Foundation
+version 2.1 of the License.
+
+This library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with this library; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301
+USA.
+*************************************************************************/
+
+#define _GNU_SOURCE
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+
+#include <ngf/log.h>
+#include <trace/trace.h>
+
+#include "tonegend.h"
+#include "ausrv.h"
+#include "tone.h"
+#include "indicator.h"
+#include "dtmf.h"
+#include "interact.h"
+
+
+#define DISCONNECTED    0
+#define CONNECTED       1
+
+#define TRACE(f, args...) trace_write(trctx, trflags, trkeys, f, ##args)
+#define LOG_CAT "tonegen-interact: "
+
+static gboolean handle_input(GIOChannel *, GIOCondition, gpointer);
+
+/********************* Temporary *************************/
+#include "note.h"
+
+static void ringtone(struct ausrv *ausrv)
+{
+    struct ndef {int dur; int note;};
+    static struct ndef ndef[] = {
+        {32, NOTE_PAUSE}, {8, NOTE_G}, {8, NOTE_PAUSE}, {16, NOTE_Ab},
```

Comment:
If the linewise grouping is relevant, align it like this:

```
{32, NOTE_PAUSE}, {8, NOTE_G},  {8, NOTE_PAUSE},  {16, NOTE_Ab},
{8, NOTE_PAUSE},  {16, NOTE_G}, {16, NOTE_PAUSE}, {16, NOTE_G},
{8, NOTE_C},      {8, NOTE_G},  {8, NOTE_F},      {8, NOTE_G}, {8, NOTE_PAUSE},
{16, NOTE_D},
{0, 0} 
```

If the grouping is not relevant, just do one-struct-per-line, but make the numbers align:

```
{32, NOTE_PAUSE},
{ 8, NOTE_G},
{ 8, NOTE_PAUSE},
{16, NOTE_Ab},
{ 8, NOTE_PAUSE},
{16, NOTE_G},
{16, NOTE_PAUSE},
{16, NOTE_G},
{ 8, NOTE_C},
{ 8, NOTE_G},
{ 8, NOTE_F},
{ 8, NOTE_G},
{ 8, NOTE_PAUSE},
{16, NOTE_D},
{ 0, 0} 
```